### PR TITLE
SDIT-2251 Revised casenotes to-nomis sync logic

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -114,7 +114,6 @@ val buildDirectory: Directory = layout.buildDirectory.get()
 
 tasks.register("buildActivityApiModel", GenerateTask::class) {
   generatorName.set("kotlin")
-  skipValidateSpec.set(true) // TODO - turn this back on when the spec is valid again!
```
Comment (steverendell) [inline build.gradle.kts:117]: I happened to noticed some of these skips arent needed any more

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesService.kt [hunk 2]
```diff
@@ -87,17 +87,18 @@ class CaseNotesService(
...
-          dpsCaseNote.toNomisUpdateRequest(),
-        )
+        val updateCaseNoteRequest = dpsCaseNote.toNomisUpdateRequest()
+        var i = 1
+        for (dto in mappings) {
+          telemetryMap["nomisBookingId-$i"] = dto.nomisBookingId.toString()
+          telemetryMap["nomisCaseNoteId-$i"] = dto.nomisCaseNoteId.toString()
+          i++
+          nomisApiService.updateCaseNote(dto.nomisCaseNoteId, updateCaseNoteRequest)
+        }
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesService.kt:101]: one dps id may now have several corresponding nomis ids (previously merged and copied case notes)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesToNomisIntTest.kt [hunk 3]
```diff
@@ -717,29 +815,26 @@ class CaseNotesToNomisIntTest : SqsIntegrationTestBase() {
     caseNoteUuid: String = UUID.randomUUID().toString(),
     source: CaseNoteSource = CaseNoteSource.DPS,
     syncToNomis: Boolean = true,
-    legacyId: Long = 3L,
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesToNomisIntTest.kt:720]: not used

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesService.kt (hunk 2 above)
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesService.kt:101]: A few ways you could drop the i++ here...

```
mapings.forEachIndexed { i ->
  ..."nomisBookingId=${I+1}"...
}  
```

```
(1..mappings.size).forEach { i ->  }
```

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesService.kt (hunk 2 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesService.kt:101]: Ah thanks yes that would be better - will change in next PR

**Pull request merged**